src/utils.c

```diff
@@ -722,3 +724,77 @@ ts_get_reloptions(Oid relid)
 
 	return options;
 }
+
+int64
+ts_get_now_internal(const Dimension *open_dim)
+{
+	Oid dim_post_part_type = ts_dimension_get_partition_type(open_dim);
+
+	if (IS_INTEGER_TYPE(dim_post_part_type))
+	{
+		Datum now_datum;
+		Oid now_func = ts_get_integer_now_func(open_dim);
+		now_datum = OidFunctionCall0(now_func);
+		return ts_time_value_to_internal(now_datum, dim_post_part_type);
+	}
+	else
```

Comment:
nit: guess this else is not needed, because you are using return in previous block. can reduce nesting here